mod/gcforums/start.php

```diff
@@ -252,162 +263,229 @@ function render_forums($forum_guid) {
+		}
 
-		/// topic
-		foreach ($topics as $topic) {
-			
-			$topic = get_entity($topic->guid);
-			$hyperlink = "<a href='{$base_url}gcforums/topic/view/{$topic->guid}'><strong>{$topic->title}</strong></a>";
-			if (!$topic->guid) continue;
-			$query = "SELECT e.guid, ue.username, e.time_created
-						FROM {$dbprefix}entities e, {$dbprefix}users_entity ue
-						WHERE e.container_guid = {$topic->guid} AND e.owner_guid = ue.guid";
-			$replies = get_data($query);
+		$query = "SELECT * FROM elggentities e, elggentity_subtypes es WHERE e.subtype = es.id AND e.container_guid = {$forum_guid} AND es.subtype = 'hjforumtopic'";
+		$topics = get_data($query);
 
+		// sort
+		usort($topics, function($a, $b) {
+		    return $b->guid - $a->guid;
+		});
 
-			$total_replies = count($replies);
-			$topic_starter = get_user($topic->owner_guid)->username;
-			$time_posted = $replies[count($total_replies) - 1]->time_created;
-			$time_posted = date('Y-m-d H:i:s', $time_posted);
+		if (count($topics) > 0 && !$entity->enable_posting) {
+			$content .= "
+				<div class='topic-main-box'>
+					<div style='background: #e6e6e6; width:100%;' >
+						<div class='topic-header'>".elgg_echo('gcforums:translate:hjforumtopic')."
+							<div class='topic-information'>options</div>
+							<div class='topic-information'>".elgg_echo('gcforums:translate:last_posted')."</div>
+							<div class='topic-information'>".elgg_echo('gcforums:translate:replies')."</div>
+							<div class='topic-information'>".elgg_echo('gcforums:translate:topic_starter')."</div>
+						</div>";
 
-			$options = render_edit_options($topic->guid, $group_entity->guid);
-			if ($options == '') $options = '-';
-			$admin_only = (elgg_is_admin_logged_in()) ? "(guid:{$topic->guid})" : "";
-			$last_post = ($total_replies <= 0) ? $last_post = elgg_echo('gcforums:no_posts') : "<div>{$replies[$total_replies - 1]->username}</div> <div>{$time_posted}</div>";
 
-			$content .= "
-			<div class='topic-info-header'>
-				<div class='topic-description'>{$hyperlink} {$admin_only}</div>
-				<div class='topic-options-edit'>{$options}</div>
-				<div class='topic-options'>{$last_post}</div>
-				<div class='topic-options'>{$total_replies}</div>
-				<div class='topic-options'>{$topic_starter}</div>
-			</div>";
-		}
+			/// topic
+			foreach ($topics as $topic) {
+				
+				$topic = get_entity($topic->guid);
+				$hyperlink = "<a href='{$base_url}gcforums/topic/view/{$topic->guid}'><strong>{$topic->title}</strong></a>";
+				if (!$topic->guid) continue;
+				$query = "SELECT e.guid, ue.username, e.time_created
+							FROM {$dbprefix}entities e, {$dbprefix}users_entity ue
+							WHERE e.container_guid = {$topic->guid} AND e.owner_guid = ue.guid";
+				$replies = get_data($query);
 
-		$content .= "</div> </div> </p> <br/>";
-	}
 
+				$total_replies = count($replies);
+				$topic_starter = get_user($topic->owner_guid)->username;
+				$time_posted = $replies[count($total_replies) - 1]->time_created;
```

Comment:
Isn't $total_replies an int at this point since it's the size of replies? I'm guessing this is supposed to be time created of the last post, so
```php
$replies[$total_replies - 1]->time_created;